### PR TITLE
Add react-refresh-typescript to documentation

### LINKED ISSUE #34: Support typescript loader ?
Does this plugin force me to use babel ? I'm developing react project while using only typescript loader in webpack. Can i use your awesome tool for my project or maybe you have plans to support other loaders ?

### LINKED ISSUE #211: Support TypeScript loader!
facebook/react#19914

Please update the README after it get merged and a new version of react-refresh is released!

### PR DESCRIPTION
**Pull request by Jack-Works**
close #211, close #34

### CODE REVIEW

**Comment by pmmmwh** [comment]
Hey - thanks for the PR! I'm open to this change, but I want to further formulate the narrative a bit that this is not officially maintained by the React team (not to disregard your work, but for some people who work in corps or cares about being purely React they would need that information) - can you add that in?

**Review by pmmmwh — APPROVED** [review]
LGTM - can you please (if you haven't) test run the example and make sure that it works, and also would recover from errors?

(I'll do it when I merge the PR but just want to be double sure)

**Comment by Jack-Works** [comment]
I've tried hmr works but not error recovery

**Comment by pmmmwh** [comment]
I'll further test this out a bit now that we have all the tests back up 😃

**Comment by Jack-Works** [comment]
hi, any progress?

**Comment by pmmmwh** [comment]
Setting up tests to run for this scenario now - if they all pass I'll merge this in.

**Review by hg-pyun — APPROVED** [review]
LGTM. This is the solution I was looking for.

File: README.md [hunk 1]
```diff
@@ -116,6 +118,66 @@ module.exports = {
...
+      {
+        test: /\.[jt]sx?$/,
+        exclude: /node_modules/,
+        use: [
+          // ... other loaders
+          {
+            loader: require.resolve('ts-loader'),
+            options: {
+              getCustomTransformers: () => ({
+                before: [ReactRefreshTypeScript()],
```
Comment (hg-pyun) [inline README.md:164]: Why doesn't use this? 
```js
before: isDevelopment ? [ReactRefreshTypeScript()] : []
```

**Comment by Jack-Works** [comment]
rebased and resolve reviews

**Comment by hg-pyun** [comment]
@pmmmwh Please Review 👍

**Review by simPod — APPROVED** [review]

**Comment by yoruvo** [comment]
Hi,

just dropping in to say that this approach fixed my problem of being unable to use this plugin with TypeScript.

My single gripe is that you need to `require("react-refresh-typescript").default` instead of just `require("react-refresh-typescript")`. That seems to oppose the way other plugins are written. This is obviously a very minor gripe though.

Also, the readme change doesn't seem to mention that CommonJS isn't supported as tsconfig target.

**Comment by Jack-Works** [comment]
@yoruvo I wrote no commonjs support in my repo maybe I should add it here too. Nice catch, generally you don't need to feed commonJS to webpack so you can overwrite your settings in your ts-loader

**Comment by Jack-Works** [comment]
@yoruvo In 1.1.1 you can require it directly without `.default`

**Comment by hg-pyun** [comment]
@Jack-Works @pmmmwh Is it ready?

**Comment by Jack-Works** [comment]
It's production ready in our app. You can install and use it without this PR, this is just a document PR

**Comment by pmmmwh** [comment]
Hi, I'm very sorry that I have been dragging this for such a long time.

I have a partially working test setup for this and I really want this to pass all conformance tests before officially recommending it. I'll see if I can whip up a branch to track the status of that this week.

**Review by pmmmwh — APPROVED** [review]

**Pull request merged**